app/pages/01_Explore.py

```diff
@@ -64,19 +140,84 @@ def _upload_to_api(file):
             n_rows = result["n_rows"]
             n_cols = result["n_cols"]
 
-            # Update session state and rerun
             st.session_state["dataset_choice"] = table_name
             st.session_state["uploader_nonce"] = nonce + 1
             st.session_state["flash"] = (
                 f"Ingested **{dataset_id}** as `{table_name}` ({n_rows}×{n_cols})."
             )
             st.rerun()
 
-    elif uploaded.name.lower().endswith(".zip"):
-        st.warning(
-            "ZIP upload not yet implemented via API. Extract and upload CSV directly."
+    elif lower_name.endswith(".zip"):
```

Comment:
Careful checking if a zip file is actually a zip file with `.endswith(".zip")`, what if someone just renames .csv to end with .zip but it's not actually compressed? `zipfile.is_zipfile(tmp_path)` might be a better solution